### PR TITLE
Add node id to _cat apis

### PR DESCRIPTION
**Pull request by lmenezes**
this addresses https://github.com/elastic/elasticsearch/issues/10226 and also adds node id to _cat/segments and _cat/plugins.

/cc @drewr @javanna

### CODE REVIEW

**Comment by lmenezes** [comment]
new CLA?

File: rest-api-spec/test/cat.plugins/10_basic.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  - match:
+      $body: |
+               /^   id          .+   \n
+                    name        .+   \n
+                    component   .+   \n
+                    version     .+   \n
+                    type        .+   \n
+                    url         .+   \n
+                    description .+   \n
+               $/
```
Comment (javanna) [inline rest-api-spec/test/cat.plugins/10_basic.yaml]: thanks!!!

File: src/main/java/org/elasticsearch/rest/action/cat/RestPluginsAction.java [hunk 2]
```diff
@@ -77,6 +77,7 @@ public RestResponse buildResponse(final NodesInfoResponse nodesInfoResponse) thr
     Table getTableWithHeader(final RestRequest request) {
         Table table = new Table();
         table.startHeaders();
+        table.addCell("id", "alias:i;default:false;desc:unique node id");
```
Comment (javanna) [inline src/main/java/org/elasticsearch/rest/action/cat/RestPluginsAction.java]: maybe we want to drop the `i` alias here and be consistent with the other cat apis? `id` seems short enough already no?

**Comment by javanna** [comment]
looks good @lmenezes I left a small comment, if you can address that I will get this in.

File: src/main/java/org/elasticsearch/rest/action/cat/RestPluginsAction.java (hunk 2 above)
Comment (lmenezes) [inline src/main/java/org/elasticsearch/rest/action/cat/RestPluginsAction.java]: sounds good :)

**Comment by lmenezes** [comment]
@javanna done. if you want me to rebase and squash it, let me know :)

**Comment by javanna** [comment]
Thanks @lmenezes merged!

**Comment by lmenezes** [comment]
@javanna great, thanks :)

**Pull request closed without merging**